beacon-chain/powchain/service.go

```diff
@@ -19,35 +19,48 @@ type Reader interface {
 	SubscribeNewHead(ctx context.Context, ch chan<- *gethTypes.Header) (ethereum.Subscription, error)
 }
 
+// Logger subscribe filtered log on the PoW chain
+type Logger interface {
+	SubscribeFilterLogs(ctx context.Context, q ethereum.FilterQuery, ch chan<- gethTypes.Log) (ethereum.Subscription, error)
+}
+
 // Web3Service fetches important information about the canonical
 // Ethereum PoW chain via a web3 endpoint using an ethclient. The Random
 // Beacon Chain requires synchronization with the PoW chain's current
 // blockhash, block number, and access to logs within the
 // Validator Registration Contract on the PoW chain to kick off the beacon
 // chain's validator registration process.
 type Web3Service struct {
-	ctx         context.Context
-	cancel      context.CancelFunc
-	headerChan  chan *gethTypes.Header
-	endpoint    string
-	blockNumber *big.Int    // the latest PoW chain blocknumber.
-	blockHash   common.Hash // the latest PoW chain blockhash.
+	ctx                 context.Context
+	cancel              context.CancelFunc
+	headerChan          chan *gethTypes.Header
+	logChan             chan gethTypes.Log
+	pubKey              string
+	endpoint            string
+	validatorRegistered bool
+	vrcAddress          common.Address
+	blockNumber         *big.Int    // the latest PoW chain blocknumber.
+	blockHash           common.Hash // the latest PoW chain blockhash.
 }
 
 // NewWeb3Service sets up a new instance with an ethclient when
 // given a web3 endpoint as a string.
-func NewWeb3Service(ctx context.Context, endpoint string) (*Web3Service, error) {
+func NewWeb3Service(ctx context.Context, endpoint string, vrcAddress common.Address, pubKey string) (*Web3Service, error) {
```

Comment:
Similar to Raul's feedback in #278. We can start another issue/PR to address that. This is a good first issue.

Reply:
I think this is going to start a bad precedent and it is trivial to add in this PR. 

Please do not kick it into another PR since your PR is the one adding too many args